sympy/utilities/lambdify.py

```diff
@@ -309,18 +312,16 @@ def lambdify(args, expr, modules=None, printer=None, use_imps=True,
     module_provided = True
     if modules is None:
         module_provided = False
-        # Use either numpy (if available) or python.math where possible.
-        # XXX: This leads to different behaviour on different systems and
-        #      might be the reason for irreproducible errors.
-        modules = ["math", "mpmath", "sympy"]
 
-        #Attempt to import numpy
         try:
             _import("numpy")
         except ImportError:
-            pass
+            # Use either numpy (if available) or python.math where possible.
+            # XXX: This leads to different behaviour on different systems and
+            #      might be the reason for irreproducible errors.
+            modules = ["math", "mpmath", "sympy"]
         else:
-            modules.insert(1, "numpy")
+            modules = ["numpy"]
```

Comment:
The point, though, is that the overarching use-case for lambdify is to use it with numpy, i.e., numpy arrays as inputs. So mpmath.zeta wouldn't work in that case anyway. 

To be honest, I think the lambdify behavior of mixing and matching functions that input/output floats (math), numpy arrays (numpy), mpf objects (mpmath), and symbolic expressions (sympy) was a bad design decision.  It leads to a lot of confusion (I've seen lots of stackoverflow questions where the person was confused because lambdified functions work with symbolic inputs, but only sometimes), and I don't see many honest use-cases for it.